.github/workflows/pr-test-lint.yml

```diff
@@ -81,14 +81,20 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-go-
 
+      # TODO: remove once the dependabot PRs for cnquery go.mod work
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+
       - name: Updates to cnquery core go.mod?
         id: cnquerygomod
-        if: github.event_name == 'pull_request' && github.actor == 'dependabot[bot]'
+        if: ((github.event_name == 'push' && github.ref_name != 'main') || github.event_name == 'pull_request') && github.actor == 'dependabot[bot]'
         run: |
           echo "COUNT_GOMOD=$(git diff-tree --name-only -r HEAD~1..HEAD | grep -c -E "^go.mod$")" >> $GITHUB_OUTPUT
 
       - name: Fix providers go.mod for dependabot PRs
-        if: github.event_name == 'pull_request' && github.actor == 'dependabot[bot]' && steps.cnquerygomod.outputs.COUNT_GOMOD == '1'
+        if: ((github.event_name == 'push' && github.ref_name != 'main') || github.event_name == 'pull_request') && github.actor == 'dependabot[bot]' && steps.cnquerygomod.outputs.COUNT_GOMOD == '1'
```

Comment:
wouldn't be `github.actor == 'dependabot[bot]' && steps.cnquerygomod.outputs.COUNT_GOMOD == '1'` enough here? COUNT_GOMOD should never be > 0 if the preconditions are not set.